common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java

```diff
@@ -153,14 +167,23 @@ public void onInfoMessage(ChatMessageReceivedEvent e) {
                 }
             }
 
-            if (hideLoginAnnouncements) {
-                if (BACKGROUND_LOGIN_ANNOUNCEMENT.matcher(msg).find()) {
+            if (loginAnnouncements != FilterType.KEEP) {
+                Matcher matcher = BACKGROUND_LOGIN_ANNOUNCEMENT.matcher(msg);
+                if (matcher.find()) {
                     e.setCanceled(true);
+                    if (loginAnnouncements == FilterType.HIDE) {
+                        return;
+                    }
+
+                    String playerName = matcher.group(3);
+                    String rank = matcher.group(2);
+
+                    sendLoginMessage(playerName, rank);
```

Comment:
The duplication here is a bit annoying. :( Don't see a very good way to address it, though. Maybe we should make sure the normal and BACKGROUND_ patterns have the same groups (you can always add a dummy `()` to pad it), and they we collect the normal and background patterns into two FilterPattern instances, and select the correct one based on the chat type? 

That sounds like a separate PR, though...

Reply:
Agreed, I think something like was discussed in DC.